src/lib/os.pl

```diff
@@ -137,7 +137,8 @@
 argv(Argv) :-
     can_be(list, Argv),
     '$argv'(Argv0),
-    (    append(_, ["--"|Argv], Argv0) ->
+    (    append(Argv1, ["--"|Argv], Argv0) ->
+         \+ member("--", Argv1),
```

Comment:
One thing that seems unusual here is the remaining `true` in this branch; maybe this is meant to be part of the condition?

Reply:
I think that might misbehave in case second  -- is the last argument and the argument to os:argv is an empty list

Reply:
I removed the unnecessary true. I testee putting the member check in the condition but then `os:argv([])` would be accepted incorrectly.

Reply:
I found a problem with the current impl, in case the args don't end on `--` `os:argv([])` is always accepted.